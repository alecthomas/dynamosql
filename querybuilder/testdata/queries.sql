SELECT * FROM gamescores WHERE UserId = :UserId
SELECT * FROM gamescores WHERE UserId = :UserId AND TopScore > :MinTopScore
SELECT * FROM gamescores WHERE UserId = :UserId AND GameTitle BETWEEN :MinGameTitle AND :MaxGameTitle AND TopScore > :MinTopScore
SELECT * FROM gamescores WHERE UserId = "103" AND GameTitle BETWEEN "Galaxy" AND "Meteor" AND TopScore > 1000
SELECT * FROM gamescores WHERE UserId = "103" AND begins_with(GameTitle, "Galaxy")
SELECT UserId, TopScore, Scores[3], Scores[3][2], Studio.Name, Studio.Location.Country, Studio.Employees[3] FROM gamescores WHERE UserId = :UserId
