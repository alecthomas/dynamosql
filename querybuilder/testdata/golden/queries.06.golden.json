{
  "Query": "SELECT document(UserId, TopScore, Scores[3], Scores[3][2], Studio.Name, Studio.Location.Country, Studio.Employees[3]) FROM gamescores WHERE UserId = :UserId",
  "Prepared": {
    "Query": {
      "AttributesToGet": null,
      "ConditionalOperator": null,
      "ConsistentRead": null,
      "ExclusiveStartKey": null,
      "ExpressionAttributeNames": {
        "#Location": "Location",
        "#Name": "Name"
      },
      "ExpressionAttributeValues": null,
      "FilterExpression": null,
      "IndexName": null,
      "KeyConditionExpression": "UserId = :UserId",
      "KeyConditions": null,
      "Limit": null,
      "ProjectionExpression": "UserId, TopScore, Scores[3], Scores[3][2], Studio.#Name, Studio.#Location.Country, Studio.Employees[3]",
      "QueryFilter": null,
      "ReturnConsumedCapacity": null,
      "ScanIndexForward": null,
      "Select": null,
      "TableName": "gamescores"
    },
    "Columns": [
      {
        "Function": {
          "Function": "document",
          "Args": [
            {
              "DocumentPath": {
                "Fragment": [
                  {
                    "Symbol": "UserId",
                    "Indexes": null
                  }
                ]
              },
              "Value": null
            },
            {
              "DocumentPath": {
                "Fragment": [
                  {
                    "Symbol": "TopScore",
                    "Indexes": null
                  }
                ]
              },
              "Value": null
            },
            {
              "DocumentPath": {
                "Fragment": [
                  {
                    "Symbol": "Scores",
                    "Indexes": [
                      3
                    ]
                  }
                ]
              },
              "Value": null
            },
            {
              "DocumentPath": {
                "Fragment": [
                  {
                    "Symbol": "Scores",
                    "Indexes": [
                      3,
                      2
                    ]
                  }
                ]
              },
              "Value": null
            },
            {
              "DocumentPath": {
                "Fragment": [
                  {
                    "Symbol": "Studio",
                    "Indexes": null
                  },
                  {
                    "Symbol": "Name",
                    "Indexes": null
                  }
                ]
              },
              "Value": null
            },
            {
              "DocumentPath": {
                "Fragment": [
                  {
                    "Symbol": "Studio",
                    "Indexes": null
                  },
                  {
                    "Symbol": "Location",
                    "Indexes": null
                  },
                  {
                    "Symbol": "Country",
                    "Indexes": null
                  }
                ]
              },
              "Value": null
            },
            {
              "DocumentPath": {
                "Fragment": [
                  {
                    "Symbol": "Studio",
                    "Indexes": null
                  },
                  {
                    "Symbol": "Employees",
                    "Indexes": [
                      3
                    ]
                  }
                ]
              },
              "Value": null
            }
          ]
        },
        "DocumentPath": null
      }
    ],
    "NamedParams": {
      ":UserId": {}
    },
    "PositionalParams": {},
    "FixedParams": {}
  }
}
